src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+        }
+        if (type.equals(Excavator.class)) {
+            machines = new ExcavatorProducer().get();
+        }
+        if (type.equals(Truck.class)) {
+            machines = new TruckProducer().get();
+        }
+        return (List<Machine>) machines;
+    }
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
    private final Map<Class<? extends Machine>, MachineProducer<? extends Machine>> producers
                 = new HashMap<>();
                 
    {
        producers.put(Bulldozer.class, new BulldozerProducer());
        producers.put(Excavator.class, new ExcavatorProducer());
        producers.put(Truck.class, new TruckProducer());
    }

    @Override
    public List<Machine> getAll(Class<? extends Machine> type) {
        MachineProducer<? extends  Machine> producer = producers.get(type);
        return producer == null ? Collections.emptyList() : new ArrayList<>(producer.get());
    }
```
this will prevent you from creating a new producer instance each time the method is invoked